skl/skl.go

```diff
@@ -84,6 +84,7 @@ type Skiplist struct {
 	hasVersions bool
 	comparator  comparatorFunc
 	OnClose     func()
+	Release     func()
```

Comment:
I think Release can be a part of OnClose itself, as we would be releasing the memory only on close.

Reply:
In case of skiplist in badger, that would mean you have two functions to run.
First one will close and delete the `.mem` file and the second one will free up the memory.

We can either change the onClose to be `[]func()` and then append both the functions or keep the `onRelease` as it is.

Reply:
@ahsanbarkati This should be called `onRelease`.

Reply:
OnClose is a general purpose function. Memtable uses it as a callback to delete WAL when skiplist is closed. I think we should have a release function.

Reply:
@jarifibrahim I saw the comments after merging the PR, may be because we commented at the same time. 

By `onRelease` do you mean that we should not expose it? or do you mean `OnRelease`?